crates/ruff_python_formatter/src/expression/expr_named_expr.rs

```diff
@@ -17,16 +18,37 @@ impl FormatNodeRule<ExprNamedExpr> for FormatExprNamedExpr {
             value,
             range: _,
         } = item;
-        write!(
-            f,
-            [
-                target.format(),
-                space(),
-                text(":="),
-                space(),
-                value.format(),
-            ]
-        )
+
+        write!(f, [target.format()])?;
+
+        let comments = f.context().comments().clone();
+        let trailing_target_comments = comments.trailing_comments(target.as_ref());
+        let leading_value_comments = comments.leading_comments(value.as_ref());
+        let dangling_item_comments = comments.dangling_comments(item);
+
+        if trailing_target_comments.is_empty() {
```

Comment:
i tried making a single `write!` call with these `if`s inside, but that doesn't work since `space()` and `soft_line_break` have different types. is there some trick to make this work? (or do we prefer multiple `write` calls?